kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -236,6 +236,7 @@ andEqualsFunctions notSimplifier = fmap mapEqualsFunctions
     , (BothT,   \_ _ _ -> constructorAndEqualsAssumesDifferentHeads, "constructorAndEqualsAssumesDifferentHeads")
     , (BothT,   \_ _ s -> overloadedConstructorSortInjectionAndEquals s, "overloadedConstructorSortInjectionAndEquals")
     , (BothT,   \_ _ s -> Builtin.Bool.termAndEquals s, "Builtin.Bool.termAndEquals")
+    , (BothT,   \_ _ s -> Builtin.Bool.termOrEquals s, "Builtin.Bool.termOrEquals")
```

Comment:
The names of these functions are misleading:

- `termAndEquals` covers `Bool` values and `andBool`,
- `termOrEquals` covers `orBool`,
- `termNotBool` covers `notBool`,

and all three cover `\and` and `\equals`. Could you clean this up as part of this pull request? I would propose:

- Split `termAndEquals` into `unifyBoolValues` and `unifyBoolAnd`.
- Rename `termOrEquals` to `unifyBoolOr`.
- Rename `termNotBool` to `unifyBoolNot`.